### PR TITLE
[USHIFT-1444] Flakey snapshotting test on metal

### PR DESCRIPTION
**Pull request by copejon**
[USHIFT-1444](https://issues.redhat.com//browse/USHIFT-1444)

Metal tests are flaking because the kickstart template is missing the thinpool creation command, an lvmd.yaml that reflects the actual storage config, and a storage class to expose the thin pool at the cluster layer.

### CODE REVIEW

**Comment by copejon** [comment]
/assign @pmtk 

cc @dhellmann

File: test/kickstart-templates/kickstart.ks.template [hunk 1]
```diff
@@ -47,6 +48,37 @@ if [ -n "REPLACE_PUBLIC_IP" ]; then
     echo "    - REPLACE_PUBLIC_IP" >>/etc/microshift/config.yaml
 fi
 
+# Set the /etc/microshift/lvmd.yaml file to expose the LVM configuration
```
Comment (dhellmann) [inline test/kickstart-templates/kickstart.ks.template:51]: Would it make sense to do this part inside the test? I could anticipate other tests that would want other contents for lvmd.yaml, for example, or different storage classes.

**Review by dhellmann — COMMENTED** [review]
/test microshift-metal-tests

File: test/kickstart-templates/kickstart.ks.template [hunk 2]
```diff
@@ -47,6 +48,37 @@ if [ -n "REPLACE_PUBLIC_IP" ]; then
...
+  name: thin
+parameters:
+  csi.storage.k8s.io/fstype: xfs
+  "topolvm.io/device-class": "ushift-thin"
+provisioner: topolvm.io
+reclaimPolicy: Delete
+volumeBindingMode: WaitForFirstConsumer
+allowVolumeExpansion: true
+EOF
+
```
Comment (ggiguash) [inline test/kickstart-templates/kickstart.ks.template:81]: Same here. This should be part of the test.

File: test/kickstart-templates/kickstart.ks.template [hunk 3]
```diff
@@ -28,6 +28,7 @@ part /boot --fstype=xfs --asprimary --size=800
 part pv.01 --grow
 volgroup rhel pv.01
 logvol / --vgname=rhel --fstype=xfs --size=REPLACE_LVM_SYSROOT_SIZE --name=root
+logvol none --thinpool --size=REPLACE_LVM_THINPOOL_SIZE --fstype=None --name=ushift-thin --vgname=rhel
```
Comment (ggiguash) [inline test/kickstart-templates/kickstart.ks.template:31]: This code can also be added in the suite setup and cleaned in the teardown. Something like:

```
Config LVM Pool
    [Documentation]    Create LVM Thin Pool
    ${stdout}    ${rc}=    Execute Command
    ...    lvcreate --type thin-pool --thinpool ushift-thin --size 1024 rhel
    ...    sudo=True    return_rc=True
    Should Be Equal As Integers    0    ${rc}

Delete LVM Pool
    [Documentation]    Delete LVM Thin Pool
    ${stdout}    ${rc}=    Execute Command
    ...    lvremove -f -y lvremove rhel/ushift-thin
    ...    sudo=True    return_rc=True
    Should Be Equal As Integers    0    ${rc}
```

**Comment by ggiguash** [comment]
@copejon , please see the https://github.com/openshift/microshift/pull/2073/commits/b6c5b2ce36a6ac43eb3cba490caf77406defb936 commit for deleting the lvmd.yaml file. It's necessary to leave the cluster in a consistent state.

File: test/kickstart-templates/kickstart.ks.template (hunk 3 above)
Comment (copejon) [reply test/kickstart-templates/kickstart.ks.template:31]: +1 I'll extract thinpool handling from the kickstart and handled it in the test suite.

cc @dhellmann

**Pull request closed without merging**